pkg/api/routes.go

```diff
@@ -104,9 +112,22 @@ func (rh *RouteHandler) SetupRoutes() {
 			rh.ListRepositories).Methods("GET")
 		g.HandleFunc("/",
 			rh.CheckVersionSupport).Methods("GET")
+		g.HandleFunc("/debug/pprof/",
+			pprof.Index).Methods("GET")
+		g.HandleFunc("/debug/pprof/cmdline",
+			pprof.Cmdline).Methods("GET")
+		g.HandleFunc("/debug/pprof/profile",
+			pprof.Profile).Methods("GET")
+		g.HandleFunc("/debug/pprof/symbol/",
+			pprof.Symbol).Methods("GET")
+		g.HandleFunc("/debug/pprof/trace",
+			pprof.Trace).Methods("GET")
 	}
 	// swagger docs "/swagger/v2/index.html"
 	rh.c.Router.PathPrefix("/swagger/v2/").Methods("GET").Handler(httpSwagger.WrapHandler)
+	// Zot Search Extension Router
+	rh.c.Router.PathPrefix("/query").Methods("GET", "POST").Handler(rh.searchHandler())
```

Comment:
Since extensions are optional, this should also be wrapped in some if <cond>

Reply:
And these extensions are protected behind authentication, so make sure the test cases ensure that.

Reply:
Yes, I have already added test cases for it

Reply:
> Since extensions are optional, this should also be wrapped in some if

Ok